src/clj/fluree/db/query/api.cljc

```diff
@@ -146,16 +157,31 @@
                  parse-t-val)]
       [alias nil])))
 
+(defn load-aliased-rule-dbs
+  [conn rule-sources]
+  (go-try
+    (loop [[rule-source & rule-sources] rule-sources
+           rule-results []]
+      (when-let [rule-source (first rule-sources)]
+        (if rule-source
+          (into rule-results
+                (if (string? rule-source)
+                  (ledger/-db (<? (jld-ledger/load conn rule-source)))
+                  rule-source))
+          (recur rule-sources rule-results))))))
+
 (defn load-alias
   [conn alias t opts]
   (go-try
     (try*
       (let [[alias explicit-t] (extract-query-string-t alias)
-            address (<? (nameservice/primary-address conn alias nil))
-            ledger  (<? (jld-ledger/load conn address))
-            db      (ledger/-db ledger)
-            t*      (or explicit-t t)]
-        (<? (restrict-db db t* opts)))
+            address      (<? (nameservice/primary-address conn alias nil))
+            ledger       (<? (jld-ledger/load conn address))
+            db           (ledger/-db ledger)
+            t*           (or explicit-t t)
+            rule-sources (<? (load-aliased-rule-dbs conn (:rule-sources opts)))
+            opts*        (assoc opts :rule-sources rule-sources)]
```

Comment:
Because `rule-sources` is already part of the opts that are being provided, it feels odd to also be providing that as an additional argument. I think I can just do the `load-aliased-rule-dbs` inside `restrict-db` and that might be cleaner.